k-distribution/include/kframework/builtin/domains.md

```diff
@@ -1115,43 +1116,42 @@ The remainder of this section consists of an implementation in K of the
 operations listed above.
 
 ```k
+module BOOL-KORE [kore, symbolic]
+  imports BOOL-COMMON
+
   rule notBool true => false
   rule notBool false => true
 
-  rule true andBool B:Bool => B:Bool
-  rule B:Bool andBool true => B:Bool
-  rule false andBool _:Bool => false
-  rule _:Bool andBool false => false
-
-  rule true andThenBool K::Bool => K
-  rule K::Bool andThenBool true => K
-  rule false andThenBool _ => false
-  rule _ andThenBool false => false
+  rule true andBool B:Bool => B:Bool [simplification]
+  rule B:Bool andBool true => B:Bool [simplification]
+  rule false andBool _:Bool => false [simplification]
+  rule _:Bool andBool false => false [simplification]
```

Comment:
Why do we mark them _all_ as simplification rules? I guess this is so that we will always optimistically try them, allowing any of them to apply? My intuition would be to provide a set of complete rules for booleans (non simplification), then add simplification rules as well as needed.

So maybe we have a sepraate module with a set of complete and mutually exclusive rules for boolean (for concrete simplifications)?

```
rule true andBool B => B
rule false andBool _ => false
rule notBool true => false
rule notBool false => true
rule true orBool _ => true
rule false orBool B => B
....
```

Then have this separate module with _all_ the rules as simplificatino rules.

Maybe there should also be a way for the user to declare that this set of rules are _confluent_, so that the booster can just assume that?

Reply:
That would be possible, too. I don' thave a full set of measurements for it but I tested the performance for a variant where all rules that focus on the 2nd argument were declared `[simplification]`, leaving the other rules (the ones you quote above, focussing on the 1st arg) as they were.

OTOH all these rules become obsolete as soon as a backend with an implemented hook is used. Therefore I kept all rules together in one module and made them all `[simplification]` for homogenous appearance.
If the variant with both rules and defining equations is desired, I can change the code.